Products/Zuul/catalog/model_catalog.py

```diff
@@ -263,49 +261,66 @@ def add_model_update(self, object_update):
         if idxs: # make sure mandatory idxs are always sent
             idxs.update( MANDATORY_FIELDS ) # Mandatory fields
 
-        model_update = IndexUpdate(object_update.obj, op=op , idxs=idxs, uid=uid)
-        self.pending_updates[object_update.uid] = model_update
-        del object_update # Make sure we dont keep references to the object
+        # store the ObjectUpdate. Defer creating the IndexUpdate
+        # until the end of the transaction or before a dirty search
+        self.pending_updates[object_update.uid] = ObjectUpdate(object_update.obj, op=op , idxs=idxs)
```

Comment:
That `del` is technically still valuable, although extra credit. But it's possible for that to leak. It should have been in the caller of this method, though.